### PR TITLE
stylized alert with better copy and logout on unauthorized

### PR DESCRIPTION
**Pull request by will-weiss**
image

### CODE REVIEW

File: src/background/responders.ts [hunk 1]
```diff
@@ -50,12 +50,51 @@ function updateActiveFeedback(state: StoreState, callback: (feedbackTarget: Feed
   return updateFeedback(state, target, callback(target))
 }
 
+const standardAlertEnding =
```
Comment (shachar-langer) [inline src/background/responders.ts:53]: I think saving `alertHTML` instead of alert strings is a risk we shouldn't take. At the moment, the `alertHTML` is completely handled by the client, but if one day we decide to fetch it from the server, it becomes a security risk. We can add the "support" message for every alert or save a "shouldShowSupportMessage" flag in the state.

File: src/background/responders.ts [hunk 2]
```diff
@@ -64,10 +103,10 @@ export const responders: Responders<Action> = {
     return { auth: { state: 'authenticated', user: { photoUrl } } }
   },
   authenticationFailure(state, { error }): Partial<StoreState> {
-    return { alert: error.message, auth: { state: 'auth_failed' } }
+    return { alertHtml: error.message, auth: { state: 'not_authed' } }
```
Comment (shachar-langer) [inline src/background/responders.ts:106]: Why did you remove the `auth_failed` state?

File: src/tests/integration/steps/posting-feedback.ts [hunk 3]
```diff
@@ -3,10 +3,11 @@ import * as fetchMock from 'fetch-mock'
 import { Mocks } from '../mocks'
 import { ensureActiveTab, ensureActiveFeedbackTarget } from '../../../selectors'
 
-export function postingFeedback(mocks: Mocks): void {
-  describe(`post feedback`, () => {
+export function postingFeedback(mocks: Mocks, opts: { result: 'success' | 'unauthorized' }): void {
```
Comment (shachar-langer) [inline src/tests/integration/steps/posting-feedback.ts:6]: Posting feedback can also fail due to problems with Twitter, problems with the server or bad connection. Can you please add a test to make sure that we stay authenticated in this case?

File: src/tests/integration/unauthorized.test.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  const mocks = createMocks()
+
+  runBackground(mocks, { allowActionFailure: true })
+  mountPopup(mocks, { handle: 'exists' })
+  signInViaTwitter(mocks)
+  authenticateViaTwitter(mocks)
+  onceAuthenticated(mocks)
+  captureFirstScreenshot(mocks)
+  feedbackEditing(mocks)
+  postingFeedback(mocks, { result: 'unauthorized' })
```
Comment (shachar-langer) [inline src/tests/integration/unauthorized.test.ts:21]: Sweet

File: src/tests/integration/steps/posting-feedback.ts (hunk 3 above)
Comment (will-weiss) [reply src/tests/integration/steps/posting-feedback.ts]: added

File: src/tests/integration/unauthorized.test.ts (hunk 4 above)
Comment (will-weiss) [reply src/tests/integration/unauthorized.test.ts:21]: 🙌

File: src/background/responders.ts (hunk 2 above)
Comment (will-weiss) [reply src/background/responders.ts]: The `not_authed` state represents it well. You're not authed and not authenticating. We just need to also display an alert, which this does.

File: src/background/responders.ts (hunk 1 above)
Comment (will-weiss) [reply src/background/responders.ts]: great idea, done.

File: src/background/responders.ts [hunk 5]
```diff
@@ -115,9 +154,9 @@ export const responders: Responders<Action> = {
       }
     })
   },
-  fetchHandleFailure(state, { tabId, domain, error }): Partial<StoreState> {
+  fetchHandleFailure(state, { tabId, domain, failure }): Partial<StoreState> {
     return {
-      alert: `Failed to set handle: ${error}`,
+      ...handleFailure(failure),
```
[the remark this answers is not in the document]
Comment (will-weiss) [reply src/background/responders.ts]: added a test

File: src/background/responders.ts [hunk 6]
```diff
@@ -50,12 +50,51 @@ function updateActiveFeedback(state: StoreState, callback: (feedbackTarget: Feed
...
+      return {
+        auth: { state: 'not_authed' },
+        alertHtml: 'Your session ended. Please log in to try again.',
+      }
+    }
+    case 'service down': {
+      return { alertHtml: 'Twitter appears to be down. Please try again later.' }
+    }
+    case 'server error': {
+      return { alertHtml: 'It looks like our se Please try again later.' }
```
[the remark this answers is not in the document]
Comment (will-weiss) [reply src/background/responders.ts]: done

**Review by shachar-langer — APPROVED** [review]
